From cf6c91350fc839b269d421c819ecc3b38d029e80 Mon Sep 17 00:00:00 2001
From: Kevin Boulain <kevinboulain@gmail.com>
Date: Tue, 23 Aug 2022 13:57:08 +0800
Subject: [PATCH] drm/i915: Fix errors on pipe A after hibernation

When booting after suspending to disk, the screen would stay black and
other consoles would spam:
  [drm] *ERROR* Fault errors on pipe A: 0x00000080

This has been introduced in
https://cgit.freedesktop.org/drm-tip/commit/drivers/gpu/drm/i915?id=39a2bd34c933b00f7c7ada923c212b3ff826fb5d
And has been the case for all kernel versions since then (starting with
5.18). For more information, see
https://gitlab.freedesktop.org/drm/intel/-/issues/6506
---
 drivers/gpu/drm/i915/display/intel_dpt.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/display/intel_dpt.c b/drivers/gpu/drm/i915/display/intel_dpt.c
index 775780adada4..a84551deedcc 100644
--- a/drivers/gpu/drm/i915/display/intel_dpt.c
+++ b/drivers/gpu/drm/i915/display/intel_dpt.c
@@ -79,9 +79,6 @@ static void dpt_bind_vma(struct i915_address_space *vm,
 {
 	u32 pte_flags;
 
-	if (vma_res->bound_flags)
-		return;
-
 	/* Applicable to VLV (gen8+ do not support RO in the GGTT) */
 	pte_flags = 0;
 	if (vm->has_read_only && vma_res->bi.readonly)
-- 
2.25.1

